### DISCUSSION TITLE
deduping repeater relations

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#11884] [Help]
### Package

Form builder

### Package Version

3

### How can we help you?

I have a repeater field that uses a belongsToMany relationship:

```php


                Repeater::make('products')
                    ->relationship()
                    ->schema([
                        TextInput::make('product_number')
                            ->required(),

                        TextInput::make('serial_number')
                            ->required(),
                    ])
                    ->defaultItems(0)
                    ->addable(false)
                    ->deletable(false)
                    ->reorderable(false)
                    ->visible(function ($state) {
                        return !empty($state);
                    }),
```

it works, fine, but i am wondering how i can dedupe the products so they are only created once in the products table

ideally i would like to check if a product already exists using the product_number and serial_number, and if it already exists in the product table then just use that record rather than creating a new one

i could not find how to do this in the docs, is this possible?

**Comment by micraux** [comment]
Hello,

You have 3 options.

Take a look at [https://filamentphp.com/docs/3.x/forms/fields/repeater#distinct-state-validation](https://filamentphp.com/docs/3.x/forms/fields/repeater#distinct-state-validation)
